### PR TITLE
2156 tupaia charts refactor

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:
- Add themeing to chart components in ui-components
- Replace web-frontend chart components with ui-components chart components
- Add table flippa to web-frontend
- Tweak and refactor styles in web-front end charts to improve consistency

---

### Screenshots:
see issue

### CODE REVIEW

File: packages/ui-components/src/components/Chart/CartesianChart.js [hunk 1]
```diff
@@ -213,7 +213,7 @@ export const CartesianChart = ({ viewContent, isEnlarged, isExporting }) => {
             });
           })}
         {ReferenceLines({ viewContent, isExporting, isEnlarged })}
-        {chartType === BAR && data.length > 20 && !isExporting && (
+        {chartType === BAR && data.length > 20 && !isExporting && isEnlarged && (
```
Comment (tcaiger) [inline packages/ui-components/src/components/Chart/CartesianChart.js:216]: hide the brush in the non expanded version of the charts

**Review by edmofro — CHANGES_REQUESTED** [review]
A perfect reviewer would go through line by line and comparing the functionality between the ui-components and web-frontend versions of the dashboard items, but I haven't done that - have just reviewed the changes. Hopefully testing will catch any of that, but it will have to be very very thorough regression testing because any bugs will be hiding in the most niche use cases on Tupaia...

On that note though, it would be worth running a diff across the current version of web-frontend and the version when you created the charts in ui-components, and see what functionality might have been added or updated, and whether the change was also reflected in ui-components.

Couldn't comment inline, but wouldn't expect to see the diff on the `pg-mv-fast-refresh` submodule, perhaps need to update from dev and maybe run one of the mv scripts to make it check out the latest from the submodule.

Glad this all worked out, and we're able to drop the duplication of code!

**Comment by edmofro** [comment]
For the diff, `git diff 94d57dd0f5c0430cedca45bc0d330659548d874e packages/web-frontend/src/components/View/` should get you the right info (that commit sha is from late Jan, about when you started working on it)

File: packages/ui-components/src/components/Chart/Tooltip.js [hunk 2]
```diff
@@ -120,9 +120,12 @@ const SingleValueTooltip = ({ valueType, payload, periodGranularity, labelType }
 export const Tooltip = props => {
   const { payload, active, presentationOptions } = props;
 
-  const data = payload || []; // This is to hancle when recharts overrides the payload as null
+  const data = payload || []; // This is to handle when recharts overrides the payload as null
   const filteredPayload = data.filter(({ value }) => value !== undefined);
 
+  // Todo: lookup
+  // const filteredPayload = data.filter(({ value }) => ![null, undefined].includes(value));
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/Tooltip.js:127]: 👀

File: packages/ui-components/src/components/Chart/parseChartConfig.js [hunk 3]
```diff
@@ -16,7 +27,34 @@ export const parseChartConfig = viewContent => {
...
+  const addDefaultColors = config => addDefaultColorsToConfig(config, paletteName, chartType);
+
+  const chartConfigs = [baseConfig];
+
+  return chartConfigs
+    .map(sortChartConfigByLegendOrder)
+    .map(addDefaultColors)
+    .map(setOpacityValues)[0]; // must remove from array after mapping
+
+  // return addDefaultColorsToConfig(sortChartConfigByLegendOrder(baseConfig), paletteName, chartType);
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/parseChartConfig.js:39]: 👀

**Review by edmofro — CHANGES_REQUESTED** [review]
Couple of commented out things to check

File: packages/ui-components/src/components/Chart/Tooltip.js (hunk 2 above)
Comment (tcaiger) [reply packages/ui-components/src/components/Chart/Tooltip.js]: Thanks for checking this. I was initially struggling to understand what this mean but I finally figured it out and realised that I need to keep this change.

**Review by edmofro — APPROVED** [review]
👏

**Comment by tcaiger** [comment]
closing as all the changes are merged into `2780-new-dashboard-endpoint`

**Comment by tcaiger** [comment]
Closing as this was merged in last week's release

**Pull request closed without merging**